ovn_k8s/modes/overlay.py

```diff
@@ -247,6 +250,11 @@ def _update_vip(self, service_data, ips):
                     self._create_load_balancer_vip(service_type, gateway_ip,
                                                    ips, port, target_port,
                                                    protocol)
+            elif external_ips:
```

Comment:
in theory external ips and NodePort are not mutually exclusive.
However, in practice it is probably unlikely to see a service exposed in both ways.

I'm not sure what the correct behaviour should be here. I'm fine with your code for the time being.
Perhaps let's just add a note in a comment as a reminder for checking again in the future